elastic/apm-server
# benchmark: Add `index-benchmark-results` target

## Motivation/summary

Adds a new makefile target to the `testing/benchmark` workspace with a
set of `GOBENCH_<FLAG>` variables that can be set to index the benchmark
results to an Elasticsearch cluster that isn't the default (localhost).

To re-use the existing makefile targets that set up the Go build env and
command dependencies, a new `go.make` has been introduced which is now
included by the main `Makefile` and `testing/benchmark/Makefile`.

## Checklist



~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~

## How to test these changes

1. `docker-compose up -d && cd testing/benchmark && make`. (Creates a new deployment in ESS).
2. `make run-benchmark BENCHMARK_TIME=30s BENCHMARK_COUNT=1`
3. `make index-benchmark-results`. 
4.  Create a new Index pattern in Kibana (Stack Management > Kibana > Data View > `gobench*`)
5. Navigate to Discover and verify that the documents are indexed.

## Related issues

Closes #7856
